src/pyqasm/expressions.py

```diff
@@ -390,6 +418,15 @@ def _check_type_size(expression, var_name, var_format, base_type):
         if isinstance(expression, FunctionCall):
             # function will not return a reqd / const type
             # Reference : https://openqasm.com/language/types.html#compile-time-constants, para: 5
+            if expression.name.name in {"abs", "real", "imag", "sqrt", "sin", "cos", "tan"}:
```

Comment:
Do we not have the inverse trig functions as well like `arccos`, `arcsin`, etc.?